src/fancy-tools/rectangleRoiTool.js

```diff
@@ -0,0 +1,415 @@
+import drawLinkedTextBox from './../util/drawLinkedTextBox.js';
+import calculateSUV from '../util/calculateSUV.js';
+
+const cornerstone = external.cornerstone;
+
+const numberWithCommas = (x) => {
+  // http://stackoverflow.com/questions/2901102/how-to-print-a-number-with-commas-as-thousands-separators-in-javascript
+  const parts = x.toString().split('.');
+
+  parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+
+  return parts.join('.');
+};
+
+export default class extends baseAnnotationTool {
+  constructor (name) {
+    super({
+      name: name || 'rectangleRoi',
+      supportedInteractionTypes: ['mouse', 'touch']
+    });
+  }
+
+  /**
+   * Create the measurement data for this tool with the end handle activated
+   *
+   * @param {*} eventData
+   * @returns
+   */
+  createNewMeasurement (eventData) {
+    const goodEventData =
+      eventData && eventData.currentPoints && eventData.currentPoints.image;
+
+    if (!goodEventData) {
+      console.error(
+        `required eventData not supplieed to tool ${
+          this.name
+        }'s createNewMeasurement`
+      );
+
+      return;
+    }
+
+    return {
+      visible: true,
+      active: true,
+      color: undefined,
+      invalidated: true,
+      handles: {
+        start: {
+          x: eventData.currentPoints.image.x,
+          y: eventData.currentPoints.image.y,
+          highlight: true,
+          active: false
+        },
+        end: {
+          x: eventData.currentPoints.image.x,
+          y: eventData.currentPoints.image.y,
+          highlight: true,
+          active: true
+        },
+        textBox: {
+          active: false,
+          hasMoved: false,
+          movesIndependently: false,
+          drawnIndependently: true,
+          allowedOutsideImage: true,
+          hasBoundingBox: true
+        }
+      }
+    };
+  }
+
+  /**
+   *
+   *
+   * @param {*} element
+   * @param {*} data
+   * @param {*} coords
+   * @returns
+   */
+  pointNearTool (element, data, coords) {
+    const hasStartAndEndHandles =
+      data && data.handles && data.handles.start && data.handles.end;
+    const validParameters = hasStartAndEndHandles;
+
+    if (!validParameters) {
+      console.warn(
+        `invalid parameters supplieed to tool ${this.name}'s pointNearTool`
+      );
+    }
+
+    if (!validParameters || data.visible === false) {
+      return false;
+    }
+
+    const startCanvas = cornerstone.pixelToCanvas(element, data.handles.start);
+    const endCanvas = cornerstone.pixelToCanvas(element, data.handles.end);
+
+    const rect = {
+      left: Math.min(startCanvas.x, endCanvas.x),
+      top: Math.min(startCanvas.y, endCanvas.y),
+      width: Math.abs(startCanvas.x - endCanvas.x),
+      height: Math.abs(startCanvas.y - endCanvas.y)
+    };
+
+    const distanceToPoint = external.cornerstoneMath.rect.distanceToPoint(rect, coords);
+
+    return (distanceToPoint < 5);
```

Comment:
I don't think it's related to your changed, but I'm having a hard time grabbing and moving an existing rectangleROI using `touch`. I think this, or some logic in the dispatcher, could be the culprit. This is mostly a note to remind me to create a follow-up bug report after reviewing this PR.